homeassistant/components/heos/media_player.py

```diff
@@ -242,6 +241,9 @@ def __init__(self, player):
         current_support = [CONTROL_TO_SUPPORT[control] for control in controls]
         self._supported_features = reduce(ior, current_support, BASE_SUPPORTED_FEATURES)
 
+        if self._source_manager is None:
```

Comment:
@andrewsayre I moved this into `async_update`, because `async_added_to_hass` is called only after we ask the capability attributes.

Reply:
Just to clarify -- because we call `async_update` to get `capability_attributes` and this was blowing up on the other properties that rely on `_source_manager`?

Reply:
Yes. We wanted to get the `source_list` property.